src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F

```diff
@@ -165,285 +166,351 @@ subroutine ocn_tracer_hmix_redi_tend(meshPool, scratchPool, layerThicknessEdge,
 
       call mpas_pool_get_array(meshPool, 'nEdgesOnCell', nEdgesOnCell)
       call mpas_pool_get_array(meshPool, 'edgesOnCell', edgesOnCell)
+      call mpas_pool_get_array(meshPool, 'cellsOnCell', cellsOnCell)
       call mpas_pool_get_array(meshPool, 'edgeSignOnCell', edgeSignOnCell)
 
-      call mpas_pool_get_config(ocnConfigs, 'config_Redi_kappa',config_Redi_kappa)
-      call mpas_pool_get_config(ocnConfigs, 'config_disable_redi_horizontal_term1',config_disable_redi_horizontal_term1)
-      call mpas_pool_get_config(ocnConfigs, 'config_disable_redi_horizontal_term2',config_disable_redi_horizontal_term2)
-      call mpas_pool_get_config(ocnConfigs, 'config_disable_redi_horizontal_term3',config_disable_redi_horizontal_term3)
-
-      !
-      ! COMPUTE the extra terms arising due to mismatch between the constant coordinate surfaces and the
-      ! isopycnal surfaces.
-      !
-
-      call mpas_pool_get_field(scratchPool, 'gradTracerEdge', gradTracerEdgeField)
-      call mpas_pool_get_field(scratchPool, 'gradTracerTopOfEdge', gradTracerTopOfEdgeField)
-      call mpas_pool_get_field(scratchPool, 'gradHTracerSlopedTopOfCell', gradHTracerSlopedTopOfCellField)
-      call mpas_pool_get_field(scratchPool, 'dTracerdZTopOfCell', dTracerdZTopOfCellField)
-      call mpas_pool_get_field(scratchPool, 'dTracerdZTopOfEdge', dTracerdZTopOfEdgeField)
-      call mpas_pool_get_field(scratchPool, 'areaCellSum', areaCellSumField)
-
-      call mpas_allocate_scratch_field(gradTracerEdgeField, .true., .false.)
-      call mpas_allocate_scratch_field(gradTracerTopOfEdgeField, .true., .false.)
-      call mpas_allocate_scratch_field(gradHTracerSlopedTopOfCellField, .true., .false.)
-      call mpas_allocate_scratch_field(dTracerdZTopOfCellField, .true., .false.)
-      call mpas_allocate_scratch_field(dTracerdZTopOfEdgeField, .true., .false.)
-      call mpas_allocate_scratch_field(areaCellSumField, .true., .false.)
-
-      gradTracerEdge => gradTracerEdgeField % array
-      gradTracerTopOfEdge => gradTracerTopOfEdgeField % array
-      gradHTracerSlopedTopOfCell => gradHTracerSlopedTopOfCellField % array
-      dTracerdZTopOfCell => dTracerdZTopOfCellField % array
-      dTracerdZTopOfEdge => dTracerdZTopOfEdgeField % array
-      areaCellSum => areaCellSumField % array
-
       nCells = nCellsArray( size(nCellsArray) )
       nEdges = nEdgesArray( size(nEdgesArray) )
 
-      !$omp do schedule(runtime)
+      allocate(minimumVal(nTracers))
+      allocate(redi_term2_edge(nTracers,nVertLevels,nEdges))
+      allocate(redi_term3_topOfCell(nTracers,nVertLevels+1,nCells))
+
+      nCells = nCellsArray(1)
+      nCellsP1 = nCellsArray( size(nCellsArray) ) + 1
+      redi_term2(:,:,:) = 0.0_RKIND
+      redi_term2_edge(:,:,:) = 0.0_RKIND
+      redi_term3_topOfCell(:,:,:) = 0.0_RKIND
```

Comment:
@vanroekel I think `redi_term2_edge` and `redi_term3_topOfCell` do not need to be zeroed out, because they are not incremented upon. Do you agree? I will keep these zero lines for `redi_term1`, `redi_term2`, `redi_term3` because those are incremented, but make them thread safe.

Reply:
@mark-petersen -- yes I agree.  Thanks for catching that